### PR TITLE
Fix trusted content empty string bug

### PR DESCRIPTION
**Pull request by rondale-sc**

### CODE REVIEW

**Comment by rondale-sc** [comment]
This fixes: https://github.com/glimmerjs/glimmer-vm/issues/807

The TL;DR is 

When in serialization mode we notice an empty string inside trusted content (either `{{{''}}}` or `{{<someHTMLSafeString>}}` rather than do nothing we send an empty comment node `<!-- % % -->` which in the rehydration builder we check for.  If found in the rehydration builder we remove it and advance the candidate.  

I found that since when this happens the `newBounds` (a type of Bounds) is set its `first` and `last` are the same element.  To ensure this doesn't error when `__closeBlock` is called I used `this.clearMismatch` to fudge it along.

Please let me know if this isn't the right approach.  :beers:

**Comment by rondale-sc** [comment]
Rebased after prettier commit 👍

**Comment by rwjblue** [comment]
Whoops! Sorry about that!
